### PR TITLE
adding CLI functionality

### PR DESCRIPTION
**Pull request by lesliepassante**
- refactors code
- adds initial tests
- e2e functionality works now

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+   *
+   * return story.match(/{{[{]?(.*?)[}]?}}/g).map(word => word.replace(/({{)|(}})/g, ''));
+   *
+   * ... it would throw a `TypeError` if the story did not contain any variables. maybe
+   * we use the line of code above to start with, and then our unit tests will help us
+   * catch the bug and we refactor to the version below.
+   *
+   */
+
+  const variables = story.match(/{{[{]?(.*?)[}]?}}/g);
```
Comment (4kategreen) [inline lib/index.js]: we should assume they won't know the regex here so a comment about what it will accept would be good.

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  if (variables.length) {
+    return variables.map(word => word.replace(/({{)|(}})/g, ''));
```
Comment (4kategreen) [inline lib/index.js]: again here with the regex

File: lib/config/index.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (4kategreen) [inline lib/config/index.json]: we should test for valid json. still type checking, but good practice to demonstrate.

File: lib/stories/disney.hbs [hunk 4]
```diff
@@ -0,0 +1 @@
+Last month, I went to Disney World with {{name}}. We traveled for {{number}} by {{vehicle}}. Finally, we got there and it was very {{adjective}}. There were {{adjective}} people {{verb}} everywhere. There were also people dressed up in {{animal}} costumes. I wish it had been more {{adjective}}, but we {{verb}} anyway. We also went on some {{adjective}} rides, called "Magic {{noun}}". {{name}} nearly fell off a ride and had to be {{verb}}. Later we went to the hotel and {{verb}}. Next year, I want to go to {{place}}, where we can {{verb}}.
```
Comment (4kategreen) [inline lib/stories/disney.hbs]: <3

**Comment by lesliepassante** [comment]
pushed new code!

**Pull request merged**